plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/exporter/DataExporterJSON.java

```diff
@@ -181,26 +164,73 @@ public void exportFooter(DBRProgressMonitor monitor) throws IOException
         out.write("\n");
     }
 
-    private void writeTextCell(@Nullable String value)
-    {
+    private void writeTextCell(@Nullable String value) {
         if (value != null) {
             getWriter().write("\"" + JSONUtils.escapeJsonString(value) + "\"");
         } else {
             getWriter().write("null");
         }
     }
 
-    private void writeCellValue(Reader reader) throws IOException
-    {
+    private void writeContentValue(
+        DBCSession session,
+        DBCResultSet resultSet,
+        DBDContent content
+    ) throws DBCException, IOException {
+        // Content
+        // Inline textual content and handle binaries in some special way
+
+        try {
+            DBDContentStorage cs = content.getContents(session.getProgressMonitor());
+            if (cs != null) {
+                if (ContentUtils.isTextContent(content)) {
+                    writeClob(content, cs);
+                } else {
+                    writeBlob(cs);
+                }
+            }
+        } finally {
+            DTUtils.closeContents(resultSet, content);
+        }
+    }
+
+    private void writeClob(
+        DBDContent content,
+        DBDContentStorage cs
+    ) throws IOException {
+        if (!exportJsonAsString && ContentUtils.isJsonContent(content)) {
+            try (Reader in = cs.getContentReader()) {
+                writeCellValue(in, false);
+            }
+        } else {
+            getWriter().write("\"");
+            try (Reader in = cs.getContentReader()) {
+                writeCellValue(in, true);
+            }
+            getWriter().write("\"");
+        }
```

Comment:
I'd do this instead:
```suggestion
        try (Reader in = cs.getContentReader()) {
            if (!exportJsonAsString && ContentUtils.isJsonContent(content)) {
                writeCellValue(in, false);
            } else {
                getWriter().write("\"");
                writeCellValue(in, true);
                getWriter().write("\"");
            }
        }
```